### ISSUE TITLE
Invalid layer name generated due to whitespace or non-ASCII characters

### ISSUE THREAD

**Issue by jodygarnett** [GeoCat/qgis-bridge-plugin#54]
Layers are not correctly published when layer name contains blank spaces or non-ASCII chars

**Comment by volaya** [comment]
This is a known issue, and we decided to show a warning in that case. Are you not seeing it when you publish your layers?

**Comment by volaya** [comment]
I correct myself: the warning is in the report dialog, not before publishing

However, I cannot reproduce here, and it seems to work fine. Can you provide more details?

**Comment by jodygarnett** [comment]
I was cleaning up an internal list of tickets and moving the issues to GitHub, so I did not personally do the original testing.

**Comment by jodygarnett** [comment]
Checking my records you did this original testing .. 6 months ago.
Perhaps you fixed and did not close it?

**Comment by pvgenuchten** [comment]
the warning was displayed, in my case also an error (related to use of / in the name)

space and / are not allowed in featuretype names in wfs, one could say geoserver should be more strict not accepting them... 

```
Traceback (most recent call last): File "/Users/geocat/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/geocatbridge/publish/publishtask.py", line 128, in run self.geodataServer.publishLayer(layer, fields) File "/Users/geocat/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/geocatbridge/publish/geoserver.py", line 78, in publishLayer self.publishStyle(layer) File "/Users/geocat/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/geocatbridge/publish/geoserver.py", line 69, in publishStyle warnings = saveLayerStyleAsZippedSld(layer, styleFilename) File "/Users/geocat/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/geocatbridge/libs/bridgestyle/bridgestyle/qgis/__init__.py", line 30, in saveLayerStyleAsZippedSld z = zipfile.ZipFile(filename, "w") 
File "/Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/zipfile.py", line 1204, in __init__ self.fp = io.open(file, filemode) FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/56/43gydxj546q_64g9my_qmzr80000gn/T/geocatbridge/4d72ea2099634f1f83ce8b44b9e54bd6/Penparcau Butterfly/Survey.zip' 
```

**Comment by jodygarnett** [comment]
I have tried twice to introduce "name checking" into geoserver - but each protocol is so different in what it is allowed I was rebuffed!

My best attempt is to add name checking and just produce a report of warnings for any services where the name is invalid:

* WFS requires an alpha starting letter, followed by alphanumeric and underscore (as per XML schema requirements)
* WCS coverage have different requirements and often start with a number
* WMS has its own restrictions also ...

Now that GeoServer supports enable / disable of specific services on a layer by layer basis maybe we could 'disable' each protocol that is not supported by the provided name.

**Comment by jodygarnett** [comment]
@davidblasby ran into this again when testing vector tiles. Does qgis have have any difference between layer name and title we can work with?

**Comment by GeoSander** [comment]
Fixed in release 4.2
